app/graphql/types/query_type.rb

```diff
@@ -135,14 +136,14 @@ def digital_object(id:)
     end
 
     def facet_values(**arguments)
-      search_params = arguments[:search_params] ? arguments[:search_params].prepare : {}
+      search_params = arguments[:search_params]&.prepare || {}
 
       Hyacinth::Config.digital_object_search_adapter.search(search_params, context[:current_user]) do |solr_params|
+        solr_params.rows(0)
         solr_params.facet_on(arguments[:field_name]) do |facet_params|
-          facet_params.rows(arguments[:limit])
-          facet_params.start(arguments[:offset])
+          facet_params.rows(arguments[:limit].to_i + 1).start(arguments[:offset])
           facet_params.sort(arguments[:order_by][:field], arguments[:order_by][:direction]) if arguments[:order_by]
-          facet_params.with_statistics!
+          facet_params.filter(arguments[:facet_filter][:match_type], arguments[:facet_filter][:value]) if arguments[:facet_filter]
         end
       end
     end
```

Comment:
With the updated functionality in this method, it would be good to update `spec/requests/graphql/query/facet_values_spec.rb` to test `facet_filter`/`match_type`.

Reply:
Definitely needed some test coverage - I added jest specs and unit tests for Solr::Params to cover param construction.